src/generate_wsc.f90

```diff
@@ -54,24 +54,27 @@ subroutine generate_wsc(mol,wsc)
    call wsc%allocate(mol%n,rep)
 
 ! ------------------------------------------------------------------------
-! initialize
-! ------------------------------------------------------------------------
-   allocate( lattr(3,wsc%cells),      source = 0 )
-   allocate( dist(wsc%cells),         source = 0.0_wp )
-   allocate( trans(wsc%cells),        source = .true. )
-! ------------------------------------------------------------------------
 ! Create the Wigner-Seitz Cell (WSC)
 ! ------------------------------------------------------------------------
    wsc%at  = 0
    wsc%itbl= 0
-!$omp parallel default(none) &
-!$omp private(ii,jj,wc,c,dist,trans,t,lattr,rw) &
-!$omp shared(mol,wsc,rep) &
-!$omp shared(mindist,minpos,nmindist,nminpos)
-!$omp do schedule(dynamic)
+
+   !$omp parallel default(none) &
+   !$omp private(ii,jj,wc,c,dist,trans,t,lattr,rw) &
+   !$omp shared(mol,wsc,rep) &
+   !$omp private(mindist,minpos,nmindist,nminpos)
+
+   allocate( lattr(3,wsc%cells),      source = 0 )
+   allocate( dist(wsc%cells),         source = 0.0_wp )
+   allocate( trans(wsc%cells),        source = .true. )
+
    ! Each WSC of one atom consists of n atoms
+   !$omp do collapse(2) schedule(dynamic,32)
    do ii=1,mol%n
       do jj=1,mol%n
+         lattr = 0
+         dist = 0.0_wp
+         trans = .true.
          !if (ii.eq.jj) cycle
          ! find according neighbours
          c=0
```

Comment:
Same here

Reply:
In allocate statement, they will be initialized once when thread will start to work.

Reply:
Ah, I see these variables need to be reset.